File: frontend/src/services/deliveryService.js
import { httpGet, httpPost } from '@/api/client';
import { API } from '@/constants/apiEndpoints';
import { orderService } from '@/services/orderService';

export const deliveryService = {
  async getStatus(orderId) {
    return httpGet(`${API.DELIVERY.STATUS}/${orderId}`);
  },

  async confirmDelivery(orderId) {
    const result = await httpPost(API.DELIVERY.CONFIRM, { orderId });
    await orderService.updateStatus(orderId, 'DELIVERED').catch(() => null);
    return result;
  },
};

File: backend/src/main/java/com/agrolink/service/DeliveryService.java
package com.agrolink.service;

import com.agrolink.dto.delivery.DeliveryConfirmRequest;
import com.agrolink.dto.delivery.DeliveryStatusResponse;

public interface DeliveryService {

    DeliveryStatusResponse status(String email, Long orderId);

    DeliveryStatusResponse confirm(String email, DeliveryConfirmRequest r);
}